core/src/utils/internal/promise.ts

```diff
@@ -120,6 +171,12 @@ export const promiseFromEvent = (element: EventTarget, event: string) => {
 	};
 };
 
+/**
+ * Create a promise that resolves once a timeout has been reached.
+ *
+ * @param delay - the delay in milli seconds
+ * @returns a promise and an unsubscribe function
+ */
```

Comment:
Probably examples would be useful for these promise utilities, but maybe will see it in another PR (when the doc will be generated)